src/Agent/NewRelic/Agent/Core/Utilities/UpdatedLoadedModulesService.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using NewRelic.Agent.Configuration;
+using NewRelic.Agent.Core.DataTransport;
+using NewRelic.Agent.Core.Time;
+using NewRelic.Agent.Core.WireModels;
+
+namespace NewRelic.Agent.Core.Utilities
+{
+    public class UpdatedLoadedModulesService : DisposableService
+    {
+        private readonly IList<string> _loadedModulesSeen = new List<string>();
+        private readonly IScheduler _scheduler;
+        private readonly IDataTransportService _dataTransportService;
+        private readonly IConfigurationService _configurationService;
+        private IConfiguration _configuration => _configurationService?.Configuration;
+
+        public UpdatedLoadedModulesService(IScheduler scheduler, IDataTransportService dataTransportService, IConfigurationService configurationService)
+        {
+            _configurationService = configurationService;
+            _dataTransportService = dataTransportService;
+            _scheduler = scheduler;
+            _scheduler.ExecuteEvery(GetLoadedModules, _configuration.UpdateLoadedModulesCycle);
+        }
+
+        private void GetLoadedModules()
+        {
+            var assemblies = AppDomain.CurrentDomain.GetAssemblies()
```

Comment:
I'm still a little concerned about performance if a large number of assemblies are loaded, but I think it's ok just to keep an eye on it for now.

Reply:
If I pulled in the AgentHealthReporter I could make a few supportability metrics.  Add them to Angler so we could track them.

Reply:
I think that we already have timing code in the agent that we could leverage for this. We used them in some infinite tracing code.

Reply:
The AgentTimerService is what I'm thinking of.